src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy

```diff
@@ -0,0 +1,256 @@
+
+        then: "Check if stats for forward is available"
+        def statsData = null
+        Wrappers.wait(STATS_LOGGING_TIMEOUT, 1) {
+            statsData = otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : flow.flowId,
+                     direction: "forward"]).dps
+            assert statsData && !statsData.empty
+        }
+        and: "Check if stats for reverse is available"
+        Wrappers.wait(STATS_LOGGING_TIMEOUT, 1) {
+            statsData = otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : reversedFlow.flowId,
+                     direction: "reverse"]).dps
+            assert statsData && !statsData.empty
+        }
+        and: "Cleanup: revert system to original state"
+        flowHelperV2.deleteFlow(flow.flowId)
+        flowHelperV2.deleteFlow(reversedFlow.flowId)
+    }
+
+    def "Create two flow with server42 Rtt feature and check feature togglers"() {
+        given: "Two active neighboring switches one with server42 and one without"
+
+        def server42switches = topology.getActiveServer42Switches();
+
+        def server42switchesDpIds = server42switches.stream().map(
+                { s -> s.getDpId() }).collect(toSet());
+
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.src.dpId in server42switchesDpIds && !server42switchesDpIds.contains(it.dst.dpId)
+        }
+
+        def server42Switch = server42switches.stream().find { s -> s.getDpId() == switchPair.src.dpId }
+
+        when: "Set server42FlowRtt toggle to false"
+        disableFlowRtt()
+
+        and: "Set server42Switch FlowRtt deactivated"
+        disableFlowRttForSwitch(server42Switch)
+
+        and: "Flow for forward metric created"
+        def checkpointTime = new Date()
+        def flow = flowHelperV2.randomFlow(switchPair)
+        flowHelperV2.addFlow(flow)
+
+        and: "Reversed flow for backward metric created"
+        def reversedFlow = flowHelperV2.randomFlow(switchPair.reversed)
+        flowHelperV2.addFlow(reversedFlow)
+
+        and: "Check if stats for forward is not available"
+        otsdb.query(checkpointTime, metricPrefix + "flow.rtt",
+                [flowid   : flow.flowId,
+                 direction: "forward"]).dps
+        then: "No data points for forward flow found"
+        thrown(HttpClientErrorException)
+
+        when: "Check if stats for reverse is not available"
+        otsdb.query(checkpointTime, metricPrefix + "flow.rtt",
+                [flowid   : reversedFlow.flowId,
+                 direction: "reverse"]).dps
+        then: "No data points for reverse flow found"
+        thrown(HttpClientErrorException)
+
+        when: "We enable global toggler"
+        enableFlowRtt()
+        checkpointTime = new Date()
+        and: "Check if stats for forward is not available"
+        otsdb.query(checkpointTime, metricPrefix + "flow.rtt",
```

Comment:
this call is not too accurate, since checkpointTime was literally millisecond ago, so you are querying stats data from 'now' to 'now'

Reply:
done